docs/agented-vm-guide.md

```diff
@@ -0,0 +1,267 @@
+# Agented VM User Guide
+
+## Table of Contents
+
+<!-- toc -->
+- [Import Agented And Agentless VMs](#import-agented-and-agentless-vms)
+- [VirtualMachine CR Creation](#virtualmachine-cr-creation)
+- [ExternalNode CR Creation](#externalnode-cr-creation)
+- [ExternalEntities CR Creation](#externalentities-cr-creation)
+- [Install Antrea-Agent On Public Cloud VM](#install-antrea-agent-on-public-cloud-vm)
+  - [Installation On Linux VMs](#installation-on-linux-vms)
+  - [Installation On Windows VMs](#installation-on-windows-vms)
+- [Troubleshoot VM Agent](#troubleshoot-vm-agent)
+  - [Linux VM](#linux-vm)
+  - [Windows VM](#windows-vm)
+<!-- /toc -->
+
+Nephe supports auto-onboarding of public cloud VMs, that can run
+`antrea-agent` on VMs. When a CloudEntitySelector selector is configured to
+import agented VMs, `nephe-controller` will auto-create an ExternalNode CR
+corresponding to each VirtualMachine CR. `Antrea-Controller` watches on the
+ExternalNode CR and then creates a corresponding ExternalEntity CR. Upon
+onboarding the agented VMs, user can define Antrea NetworkPolicies using
+ExternalEntity selector.
+
+For more information about the `ExternalNode` feature, please refer to antrea
+[ExternalNode](https://github.com/antrea-io/antrea/blob/main/docs/external-node.md)
+documentation.
+
+Note: While applying antrea manifests, make sure to enable `externalnode`
+feature in the `antrea-controller` configuration. Also, make sure that VM
+can connect to the Kubernetes API server and Antrea API server.
+
+## Import Agented And Agentless VMs
+
+To import public cloud VMs that can run in agented or agentless configuration,
+the user needs to specify the `agented` flag in CloudEntitySelector CR.
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: crd.cloud.antrea.io/v1alpha1
+kind: CloudEntitySelector
+metadata:
+  name: cloudentityselector-aws-sample
+  namespace: vm-ns
+spec:
+  accountName: cloudprovideraccount-aws-sample
+  vmSelector:
+    - vmMatch:
+        - matchID: "i-0a20bae92ddcdb60b"
+        - matchID: "i-05e3fb66922d56e0a"
+        - matchID: "i-0033eb4a6c846451d"
+      agented: false
+    - vmMatch:
+        - matchID: "i-019459b33d951b62e"
+      agented: true
+EOF
+```
+
+## VirtualMachine CR Creation
+
+The account poller will poll the cloud inventory and create VirtualMachine CR.
```

Comment:
rename to nephe controller. We are not talking module level here

Reply:
Done